File: mario.c
#include <cs50.h>
#include <stdio.h>

void pyramid_left(int height, int row);
void pyramid_right(int height, int row);

int main(void)
{
    int h;
    do
    {
        h = get_int("Height: ");
    }
    while (h < 1 || h > 8);

    for (int r = 1; r <= h; r++)
    {
        // Print left part
        pyramid_left(h, r);
        
        // Print two spaces
        printf("  ");

        // Print right part
        pyramid_right(h, r);
        
        // Print newline
        printf("\n");
    }
}

void pyramid_left(int height, int row)
{
    // Print spaces
    for (int r = 1; r <= (height - row); r++)
    {
        printf(" ");
    }
    
    // Print blocks
    for (int r = 1; r <= row; r++)
    {
        printf("#");
    }
}

void pyramid_right(int height, int row)
{
    // Print blocks
    for (int r = 1; r <= row; r++)
    {
        printf("#");
    }
}

File: mario.py
from cs50 import get_int

while True:
    h = get_int("Height: ")

    if (h > 0 and h < 9):
        break

for i in range(1, h + 1):

    # Print whitespaces first
    for j in range(h - i):
        print(" ", end="")

    # Now print blocks
    for j in range(i):
        print("#", end="")

    # Print a newline
    print()